internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/copilot-cli/internal/pkg/aws/identity"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type localRunVars struct {
+	name    string
+	appName string
+	envName string
+}
+
+type localRunOpts struct {
+	localRunVars
+
+	store store
+	ws    wsWlDirReader
+	sel   wsSelector
+}
+
+func newLocalRunOpts(vars localRunVars) (*localRunOpts, error) {
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("local run"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	store := config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region))
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		store: store,
+		ws:    ws,
+		sel:   selector.NewLocalWorkloadSelector(prompt.New(), store, ws),
+	}
+	return opts, nil
+}
+
+func (o *localRunOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	return nil
+}
+
+func (o *localRunOpts) Ask() error {
+	if err := o.askWorkloadName(); err != nil {
+		return err
+	}
+	if err := o.askEnvName(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *localRunOpts) Execute() error {
+	//TODO: Get build information from the manifest and task definition for services
+	//By username:varun359
+
+	return nil
+}
+
+func (o *localRunOpts) askEnvName() error {
+	if o.envName != "" {
+		return nil
+	}
+
+	name, err := o.sel.Environment("Select an environment", "", o.appName)
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = name
+	return nil
+}
+
+func (o *localRunOpts) askWorkloadName() error {
+	if o.name != "" {
+		return nil
+	}
```

Comment:
We should validate the value of `o.name` passed by flag as well, to make sure it is a valid existed wkld name in the current workspace.